### PR TITLE
ComAssumption cleanup

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: library/lib.mli [hunk 1]
```diff
@@ -182,14 +182,16 @@ val variable_section_kind : Id.t -> Decl_kinds.binding_kind
...
-val add_section_variable : name:Id.t -> kind:Decl_kinds.binding_kind -> poly:bool -> unit
+val add_section_variable : name:Id.t -> kind:Decl_kinds.binding_kind -> unit
 val add_section_context : Univ.ContextSet.t -> unit
 val add_section_constant : poly:bool -> Constant.t -> Constr.named_context -> unit
 val add_section_kn : poly:bool -> MutInd.t -> Constr.named_context -> unit
 val replacement_context : unit -> Opaqueproof.work_list
 
 val is_polymorphic_univ : Univ.Level.t -> bool
 
+val check_same_poly : poly:bool -> unit -> unit
```
Comment (ejgallego) [inline library/lib.mli]: Would you mind adding a docstring? While I can infer what the function does the name doesn't seem very descriptive to me.

**Review by ejgallego — COMMENTED** [review]
I'm afraid I'm not familiar enough will all the subtle details involving universes and variables / contexts as to understand whether pushing the universe declaration up makes sense.

But from my limited understanding, this PR looks pretty good to me.

File: library/lib.mli (hunk 1 above)
Comment (gares) [reply library/lib.mli]: I'd put "section something" in the function name, what is unclear is the object being checked

Comment (SkySkimmer) [reply library/lib.mli]: Changed

**Comment by ejgallego** [comment]
@SkySkimmer who would be the right assignee for this PR?

**Comment by ppedrot** [comment]
@SkySkimmer How does this interfere with #10664?

**Comment by SkySkimmer** [comment]
Probably a bit in e32f9a5

**Comment by SkySkimmer** [comment]
@ppedrot this is the PR I talked to you about offline

>@SkySkimmer who would be the right assignee for this PR?

I dunno, anyone who wants it I guess.

**Comment by SkySkimmer** [comment]
@ppedrot note new commit

**Comment by SkySkimmer** [comment]
https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/787/console
~~~
┌────────────────────────┬─────────────────────────┬─────────────────────────────────────────────┬─────────────────────────────────────────────┬───────────────────────────────┬────────────────────────────┐
│                        │      user time [s]      │                 CPU cycles                  │              CPU instructions               │     max resident mem [KB]     │         mem faults         │
│                        │                         │                                             │                                             │                               │                            │
│           package_name │     NEW     OLD PDIFF   │               NEW               OLD PDIFF   │               NEW               OLD PDIFF   │        NEW        OLD PDIFF   │     NEW     OLD    PDIFF   │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│           coq-coqprime │  184.82  187.10 -1.22 % │      513700791550      519236488348 -1.07 % │      915664235698      915657656320 +0.00 % │     803512     803732 -0.03 % │     140     127   +10.24 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│             coq-sf-plf │   44.39   44.68 -0.65 % │      125151586702      125060645088 +0.07 % │      162121757788      162136020508 -0.01 % │     493328     493292 +0.01 % │      36      12  +200.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│ coq-mathcomp-odd-order │ 1425.32 1433.38 -0.56 % │     3974011708399     3995323741999 -0.53 % │     6850777396173     6850764498792 +0.00 % │    1211892    1211888 +0.00 % │     112      32  +250.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│               coq-corn │ 1481.03 1489.32 -0.56 % │     4129881705252     4155527171512 -0.62 % │     6117289049063     6117047555847 +0.00 % │     873112     873104 +0.00 % │       1      79   -98.73 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│               coq-hott │  389.28  390.60 -0.34 % │     1059978745273     1063002289370 -0.28 % │     1611757407488     1611728477151 +0.00 % │     584780     584776 +0.00 % │     457     250   +82.80 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│            coq-bignums │   68.55   68.77 -0.32 % │      188847575657      189429771957 -0.31 % │      252063655565      252101032913 -0.01 % │     495716     495672 +0.01 % │     143     143    +0.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│         coq-verdi-raft │ 1321.63 1323.83 -0.17 % │     3686527788510     3692459634545 -0.16 % │     5004433460667     5004382643163 +0.00 % │    1820960    1820832 +0.01 % │       0       0     +nan % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│          coq-fourcolor │ 2170.43 2172.90 -0.11 % │     6051718394288     6060033573793 -0.14 % │    11356646757878    11356737402306 -0.00 % │     930432     930536 -0.01 % │       2       0     +nan % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│        coq-lambda-rust │ 1527.84 1529.53 -0.11 % │     4257829102463     4258812328360 -0.02 % │     5651645582356     5651386031236 +0.00 % │    1419168    1419396 -0.02 % │      45       1 +4400.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│           coq-compcert │  773.68  774.35 -0.09 % │     2154805139649     2156910069988 -0.10 % │     2886857838388     2886739310075 +0.00 % │    1054120    1054708 -0.06 % │     312     307    +1.63 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│              coq-verdi │  119.75  119.85 -0.08 % │      332905475581      333268181136 -0.11 % │      419113030588      419188337320 -0.02 % │     560468     561088 -0.11 % │      12       9   +33.33 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│ coq-mathcomp-character │  179.89  180.04 -0.08 % │      501481530542      501630116751 -0.03 % │      666475019849      666392167184 +0.01 % │     845676     845424 +0.03 % │      18      42   -57.14 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│             coq-geocoq │ 1570.24 1571.54 -0.08 % │     4383784188932     4390015087576 -0.14 % │     6434647123156     6434840013766 -0.00 % │    1365168    1365268 -0.01 % │     339     432   -21.53 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│              coq-color │  705.03  705.52 -0.07 % │     1970155247797     1971719989967 -0.08 % │     2334531447827     2334467835941 +0.00 % │    1130644    1130672 -0.00 % │     198     154   +28.57 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│  coq-mathcomp-solvable │  188.06  188.16 -0.05 % │      524522636801      523680871065 +0.16 % │      703311268161      703259063125 +0.01 % │     774008     773984 +0.00 % │       5       4   +25.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│              coq-flocq │  245.36  245.39 -0.01 % │      681050394200      681447722997 -0.06 % │      856465067111      856463846772 +0.00 % │    1015888    1015916 -0.00 % │     284      52  +446.15 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│       coq-math-classes │  215.08  214.98 +0.05 % │      600863007660      601163218264 -0.05 % │      765870172173      765937952009 -0.01 % │     514288     514264 +0.00 % │      44      23   +91.30 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│ coq-fiat-crypto-legacy │ 6225.27 6218.64 +0.11 % │    17338171729163    17310028709246 +0.16 % │    28225615799870    28211789017469 +0.05 % │    2408980    2409104 -0.01 % │    1281    1247    +2.73 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│   coq-mathcomp-algebra │  162.42  162.20 +0.14 % │      452607057058      451786308994 +0.18 % │      563432343893      563427036458 +0.00 % │     614120     613536 +0.10 % │      14       9   +55.56 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│           coq-bedrock2 │  397.52  396.96 +0.14 % │     1105515142865     1105722361448 -0.02 % │     1821230685270     1820653118329 +0.03 % │    1918964    1930688 -0.61 % │     429     577   -25.65 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│       coq-fiat-parsers │  649.09  648.11 +0.15 % │     1814606949390     1813949126991 +0.04 % │     2733884533719     2733866666829 +0.00 % │    3197096    3205964 -0.28 % │     266     660   -59.70 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│     coq-mathcomp-field │  228.90  228.47 +0.19 % │      638209911304      636668265001 +0.24 % │      914591672665      914580177708 +0.00 % │     738920     738852 +0.01 % │       1      77   -98.70 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│        coq-fiat-crypto │ 4371.17 4362.16 +0.21 % │    12142720389185    12111881730047 +0.25 % │    19164922680325    19155790870534 +0.05 % │    2381416    2380808 +0.03 % │    1240     852   +45.54 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│            coq-coqutil │   60.56   60.38 +0.30 % │      168552176590      167710497360 +0.50 % │      204039588286      204039388042 +0.00 % │     544004     543908 +0.02 % │      53       2 +2550.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│            coq-unimath │ 4079.44 4066.69 +0.31 % │    11365987123361    11326187739396 +0.35 % │    20889509221559    20890159797688 -0.00 % │    1769532    1782284 -0.72 % │     523     290   +80.34 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│  coq-mathcomp-fingroup │   51.18   50.87 +0.61 % │      142715869415      142039471961 +0.48 % │      186184746088      186170011124 +0.01 % │     561920     562216 -0.05 % │       4       1  +300.00 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│         coq-coquelicot │   77.55   76.96 +0.77 % │      212166685498      211872809381 +0.14 % │      256944938722      256951126452 -0.00 % │     674796     674660 +0.02 % │     215      16 +1243.75 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│          coq-fiat-core │  115.14  114.10 +0.91 % │      325475779770      324905821001 +0.18 % │      405164893732      405035299488 +0.03 % │     496324     496352 -0.01 % │     638     302  +111.26 % │
├────────────────────────┼─────────────────────────┼─────────────────────────────────────────────┼─────────────────────────────────────────────┼───────────────────────────────┼────────────────────────────┤
│ coq-mathcomp-ssreflect │   40.18   39.80 +0.95 % │      110520185542      110405160034 +0.10 % │      134351579824      134357710190 -0.00 % │     520456     520464 -0.00 % │     105      23  +356.52 % │
└────────────────────────┴─────────────────────────┴─────────────────────────────────────────────┴─────────────────────────────────────────────┴───────────────────────────────┴────────────────────────────┘
~~~

**Review by ppedrot — APPROVED** [review]

**Pull request merged**